calculateSNR.m

```diff
@@ -286,86 +305,88 @@
         close(f);
 
         % ----------------------------------------------------------------
-        % zscore at target frequency and harmonics (average amp first)
-        mXavgHarmonics = getAvgHarmonics(abs(FT), cfg.idxHarmonics, cfg.binSize);
+        % Zscore at target frequency and harmonics. We will first pool
+        % (average) magnitudes across the different harmonics. To this end, we
+        % will cut segments around the target frequency and each harmonic from
+        % the long spectra. We will average these segments and from this
+        % avergae, we will extract the zscore (signal vs. noise). 
+        
+        % cut out segments of frequency bins around each frequency of interest
+        % (target and its harmonics) and average these segments 
+        mXavgHarmonics = getAvgHarmonics(mX, cfg.idxHarmonics, cfg.binSize);
 
+        % define indices of positions in the averaged segment that represent
+        % noise (note that the center position is the frequency bin where we
+        % expect signal, and noise is on either side of this bin) 
         noiseFs = [(cfg.binSize + 1 - cfg.binSize / 2 - cfg.gap) ...
                    :(cfg.binSize + 1 - 1 - cfg.gap) ...
                    (cfg.binSize + 1 + 1 + cfg.gap) ...
                    :(cfg.binSize + 1 + cfg.binSize / 2 + cfg.gap)];
 
+        % extract magnitudes of signal and noise from the average segment 
         AmpNoise = mXavgHarmonics(noiseFs, :);
         NoiseMean = mean(AmpNoise, 1);
         NoiseSD = std(AmpNoise, 0, 1);
+        % get zscore 
         targetHarmonicsZ = (mXavgHarmonics(cfg.binSize + 1, :) - NoiseMean) ./ NoiseSD;
```

Comment:
this part is almost similar to what `calculateFourier` function is doing, but you are doing it not for the target frequency but for the harmonics, right?